### PR TITLE
Users_Invitation table Migration Fix

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

Unblocks third party hosts from migrating.

## Technical Summary

Issue was flagged on [forum](https://forum.dimagi.com/t/unable-to-deploy-prod-stable-b23df94-25-apr-2024/10754) for a third party environment and locally via slack [thread](https://dimagi.slack.com/archives/C02QG63MU/p1713988073257969).

Summary of Issue:
[PR1](https://github.com/dimagi/commcare-hq/pull/34359) created a management command that imports the model `Invitation`. This management command is called in migrations `0058_createprofileanduserdata_copylocation.py` and `0059_copy_transient_invitation_supply_point.py`. A later migration `0060_invitations_addtableau_roles_and_groupids`, (related [PR](https://github.com/dimagi/commcare-hq/pull/34445)) adds two new fields `tableau_role` and `tableau_group_ids` to the `Invitation` model.

There is no issue if the environment is kept up to date with each migration. However, it is an issue if the environment is behind and is particularly a problem for third party hosts who won't know how to debug this. The issue is that `0058` and `0059` calls the management command, importing the `Invitation` model before the `tableau...` fields are updated. So at migration `0058`, the database doesn't know what to do with the `tableau...` fields that it sees in the model imported in the management command. This results in an error.

PR change:
The change in this PR catches failure of auto migration and alerts the user by directing them to the changelog. The chnagelog contains direction on how to manage the migration.

* [X] TODO: Update link to changelog once https://github.com/dimagi/commcare-cloud/pull/6277 is merged

## Feature Flag

No
### Safety story

locally tested. The management command has already been introduced and ran on Dimagi environments. The changes here are to address errors for 3rd part environments by checking migration state. No changes have been made to data aspect of the management command.

### Automated test coverage

no automated tests

### QA Plan

no QA

### Migrations

- [X] The migrations in this code can be safely applied first independently of the code

The migrations have already been done on Dimagi environments and the related code changes that depend on the migration have already been deployed.
related [PR1](https://github.com/dimagi/commcare-hq/pull/34359) [PR2](https://github.com/dimagi/commcare-hq/pull/34363)

### Rollback instructions

The [changelog](https://github.com/dimagi/commcare-cloud/pull/6277) directions depend on the change in this PR. So a reversion of this PR requires also reverting the changelog.

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/migrations/0059_copy_transient_invitation_supply_point.py [hunk 1]
```diff
@@ -1,14 +1,73 @@
...
+COUNT_ITEMS_TO_BE_MIGRATED = "SELECT COUNT(*) FROM users_invitation WHERE users_invitation.location_id is null AND users_invitation.supply_point IS NOT NULL"
+GIT_COMMIT_WITH_MANAGEMENT_COMMAND = "c3f318f328372e904f4b1db42177309211ceb332"
+AUTO_MIGRATE_ITEMS_LIMIT = 200000
+AUTO_MIGRATE_COMMAND_NAME = "copy_invitation_supply_point"
+AUTO_MIGRATE_FAILED_MESSAGE = """
+This migration cannot be performed automatically and must instead be run manually
+before this environment can be upgraded to the latest version of CommCare HQ.
+Instructions for running the migration can be found at this link:
+
+TO BE UPDATED
```
Comment (biyeun) [inline corehq/apps/users/migrations/0059_copy_transient_invitation_supply_point.py:21]: can you update this please? Thanks!

Comment (Jtang-1) [reply corehq/apps/users/migrations/0059_copy_transient_invitation_supply_point.py]: [b822aaf](https://github.com/dimagi/commcare-hq/pull/34525/commits/b822aafdd3b2f5997e8d4671cfcc2e9cab5f88e5)

File: corehq/apps/users/management/commands/copy_invitation_supply_point.py [hunk 2]
```diff
@@ -1,19 +1,53 @@
...
-                                .exclude(supply_point__isnull=True)
-                                .filter(location__isnull=True))
-        for invitation in with_progress_bar(
-                queryset_to_iterator(invitations_queryset, Invitation), invitations_queryset.count()):
-            invitation.location = SQLLocation.by_location_id(invitation.supply_point)
-            invitation.save()
+        force = options["force"]
+        status = get_migration_status(ALL_DOMAINS, MIGRATION_SLUG)
+        if status == MigrationStatus.COMPLETE and not force:
+            print("Copying supply point field to location field has already been marked as complete")
```
Comment (biyeun) [inline corehq/apps/users/management/commands/copy_invitation_supply_point.py:32]: nit
```suggestion
            self.stderr.write("Copying supply point field to location field has already been marked as complete")
```

Comment (Jtang-1) [reply corehq/apps/users/management/commands/copy_invitation_supply_point.py]: [4f69d15](https://github.com/dimagi/commcare-hq/pull/34525/commits/4f69d15d5627274c3e3905a61ed99a36beb25737)

**Review by biyeun — APPROVED** [review]

**Pull request merged**